src/modules/sensors/sensors.cpp

```diff
@@ -519,18 +534,64 @@ Sensors::adc_poll(struct sensor_combined_s &raw)
 								 ORB_PRIO_DEFAULT);
 					}
 
+				} else
 #endif
+				{
+					for (int b = 0; b < BOARD_NUMBER_BRICKS; b++) {
+
+						/* Do this once for the lowest (highest priority supply on power controller)
+						 * that is valid
+						 */
+						if (selected_source < 0 && valid_chan[b]) {
+							/* Indicate the lowest brick (highest priority supply on power controller)
+							 * that is valid as the one that is the selected source for the
+							 * VDD_5V_IN
+							 */
+
+							selected_source = b;
+
+							/* Move the selected_source to instance 0 */
+							if (_battery_pub_intance0ndx != b) {
+
+								tmp_h = _battery_pub[_battery_pub_intance0ndx];
+								_battery_pub[_battery_pub_intance0ndx] = _battery_pub[b];
+								_battery_pub[b] = tmp_h;
+								_battery_pub_intance0ndx = b;
+							}
+						}
+
+						// todo:per brick scaling
+						/* look for specific channels and process the raw voltage to measurement data */
+						if (bat_voltage_v_chan[b] == buf_adc[i].am_channel) {
+							/* Voltage in volts */
+							bat_voltage_v[b] = (buf_adc[i].am_data * _parameters.battery_voltage_scaling) * _parameters.battery_v_div;
+
+						} else if (bat_voltage_i_chan[b] == buf_adc[i].am_channel) {
+							bat_current_a[b] = ((buf_adc[i].am_data * _parameters.battery_current_scaling)
+									    - _parameters.battery_current_offset) * _parameters.battery_a_per_v;
+						}
+					}
 				}
 			}
 
-			if (_parameters.battery_source == 0 && updated_battery) {
-				actuator_controls_s ctrl;
-				orb_copy(ORB_ID(actuator_controls_0), _actuator_ctrl_0_sub, &ctrl);
-				_battery.updateBatteryStatus(t, bat_voltage_v, bat_current_a, ctrl.control[actuator_controls_s::INDEX_THROTTLE],
-							     _armed, &_battery_status);
+			if (_parameters.battery_source == 0) {
+
+				for (int b = 0; b < BOARD_NUMBER_BRICKS; b++) {
 
-				int instance;
-				orb_publish_auto(ORB_ID(battery_status), &_battery_pub, &_battery_status, &instance, ORB_PRIO_DEFAULT);
+					/* Consider the brick connected if there is a voltage */
+
+					bool connected = bat_voltage_v[b] > 1.5f;
```

Comment:
I think that's a great fundamental question for most hardware in the system. FMUv5 establishes the notion of being able to determine if a piece of hardware is expected to be in the system.

a) is present in functioning correctly. 
b) expected to be there but is not.

battery(ies) can be connected to the system and   Vbat >= Vconnected (SW) and Vbat < Vvalid (HW).  This allows us to log that fact. I would like to have a param per/brick for Vconnected

But you tell me if it make sense.